crates/swc_ecma_minifier/src/compress/pure/misc.rs

```diff
@@ -247,76 +248,73 @@ impl Pure<'_> {
 
     /// `new RegExp("([Sap]+)", "ig")` => `/([Sap]+)/gi`
     fn optimize_regex(&mut self, args: &mut Vec<ExprOrSpread>, span: &mut Span) -> Option<Expr> {
-        if args.is_empty() || args.len() > 2 {
-            return None;
+        fn valid_pattern(pattern: &Expr) -> Option<JsWord> {
+            if let Expr::Lit(Lit::Str(s)) = pattern {
+                if s.value.contains(|c: char| {
+                    // whitelist
+                    !c.is_ascii_alphanumeric()
+                        && !matches!(c, '$' | '[' | ']' | '(' | ')' | '{' | '}' | '-' | '+' | '_')
+                }) {
+                    return None;
+                }
+                if s.value.contains("\\\0") || s.value.contains('/') {
+                    return None;
+                }
+                Some(s.value.clone())
+            } else {
+                None
+            }
         }
+        fn valid_flag(flag: &Expr) -> Option<JsWord> {
+            if let Expr::Lit(Lit::Str(s)) = flag {
+                let mut set = FxHashSet::default();
+                for c in s.value.chars() {
+                    if !matches!(c, 'd' | 'g' | 'i' | 'm' | 's' | 'u' | 'y') {
```

Comment:
These flags are considered "invalid" in the absence of a polyfill, specifically regarding the sticky and unicode features. check the core-js-compat data.

```json
{
  "es.regexp.sticky": {
    "android": "49",
    "bun": "0.1.1",
    "chrome": "49",
    "chrome-android": "49",
    "deno": "1.0",
    "edge": "13",
    "electron": "0.37",
    "firefox": "3",
    "firefox-android": "4",
    "hermes": "0.3",
    "ios": "10.0",
    "node": "6.0",
    "oculus": "3.0",
    "opera": "36",
    "opera-android": "36",
    "opera_mobile": "36",
    "quest": "3.0",
    "react-native": "0.69",
    "safari": "10.0",
    "samsung": "5.0"
  }
}
```